taichi/backends/metal/codegen_metal.cpp

```diff
@@ -429,7 +429,13 @@ class KernelCodegenImpl : public IRVisitor {
 
   void visit(ReturnStmt *stmt) override {
     // TODO: use stmt->ret_id instead of 0 as index
-    emit("*{}.ret0() = {};", kContextVarName, stmt->values[0]->raw_name());
+    int idx{0};
+    for (auto &value : stmt->values) {
+      emit("{}.ret0()[{}] = {};", kContextVarName, idx, value->raw_name());
+      idx += 2;
+      // metal only support i32 array, but there are i64 slots in
```

Comment:
This comment is a bit confusing - "metal only support i32 array" is not true. We can only say `the return array of our Metal backend is of type i32`.

Reply:
done